### PR TITLE
Add PR Template

### PR DESCRIPTION
**Pull request by piyushswain**
This PR adds a new PR template for the repository

### CODE REVIEW

**Comment by DanielePalombo** [comment]
@piyushswain how is going with this Template? 🤔

**Comment by DanielePalombo** [comment]
What do you think to add a section where we can specify the severity of the COP.
ref https://github.com/solidusio/rubocop-solidus/issues/49

**Comment by piyushswain** [comment]
> What do you think to add a section where we can specify the severity of the COP. ref #49

Yes, we should look at adding a severity level to the PR template.
Will update

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+-----------------
+
+**Before submitting the PR make sure the following are checked:**
+
+* [ ] The PR relates to *only* one cop with a clear title and description.
+* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
+* [ ] Squashed related commits together.
+* [ ] Added tests.
+* [ ] Ran and ensured all tests are passing on a development environment.
+* [ ] If this is a new cop, added an entry for the cop on `/config/default.yml`
```
[the remark this answers is not in the document]
Comment (piyushswain) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: Will add another action item here for the changelog

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+**Replace this text with a summary of the changes in your PR.
+The more detailed you are, the better.**
+
+-----------------
+Severity:
```
[the remark this answers is not in the document]
Comment (piyushswain) [reply .github/PULL_REQUEST_TEMPLATE.md]: Hey this is the preview from github's markdown editor
Screenshot 2023-08-03 at 12 26 10 PM

I don't know why the file preview is a bit weird but this one looks fine in the markdown editor

**Review by MassimilianoLattanzio — APPROVED** [review]
LGTM. Thanks for the updates!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MassimilianoLattanzio) [reply .github/PULL_REQUEST_TEMPLATE.md]: After merging #51, it's better to add another action item about updating the documentation running the `bundle exec rake` task

**Pull request merged**